x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -193,9 +193,20 @@ export class MapEmbeddable
     // Passing callback into redux store instead of regular pattern of getting redux state changes for performance reasons
     store.dispatch(setOnMapMove(this._propogateMapMovement));
 
-    this._dispatchSetQuery({
-      forceRefresh: false,
-    });
+    this._dispatchSetQuery({ forceRefresh: false });
+    this._subscriptions.push(
+      shouldFetch$<MapEmbeddableInput>(this.getUpdated$(), () => {
+        return {
+          ...this.getInput(),
+          filters: this._getFilters(),
```

Comment:
It looks like these filters are only different from the ones on input due to filtering out disabled filters & those that are `controlledBy` this map.

It seems like disabled filters are already filtered out by `onlyDisabledFiltersChanged` - I wonder if we could get rid of this special treatment by adding a `excludeControlledBy` argument to `FilterCompareOptions`?

Reply:
Interesting idea and something to consider for future refactors.